### DISCUSSION TITLE
More HTTP methods?

### DISCUSSION THREAD

**Discussion by lilingxi01** [trpc/trpc#4579] [Q&A]
Just curious on the reason why tRPC does not support HTTP methods other than GET and POST.

I understand that other HTTP methods are basically having the same functionality as GET and POST, but sometimes we might still want methods like PATCH and DELETE for API standardization. So I wonder if it is hard to accomplish within tRPC current architecture, or if there is a reason for deprecating them.

(I have tried my best to search for a similar question, but I could not find one)

**Comment by drago1520** [comment]
bump

**Comment by gitsult4n** [comment]
tRPC is RPC, not REST — a procedure is identified by its name in the path/body, not by HTTP verb + resource. The GET/POST split maps to semantics: `query` = GET (side-effect free, so it's cacheable and batchable by the browser/link), `mutation` = POST (has a side effect). PATCH/DELETE would be redundant here since routing is already done by procedure name, not by verb.

If you actually need REST semantics (e.g. a public API with a verb-per-resource contract for external consumers), use `trpc-to-openapi` — it maps tRPC procedures to the full REST verb set (GET/POST/PUT/PATCH/DELETE) for that use case, while your internal tRPC client stays RPC-style.
